tests/e2e/testcases/upgrade/suite.go

```diff
@@ -0,0 +1,174 @@
+
+		containsVersion1 = strings.Contains(output, subnetEVMVersion1)
+		containsVersion2 = strings.Contains(output, subnetEVMVersion2)
+		gomega.Expect(containsVersion1).Should(gomega.BeFalse())
+		gomega.Expect(containsVersion2).Should(gomega.BeTrue())
+
+		commands.DeleteSubnetConfig(subnetName)
+	})
+
+	ginkgo.It("can upgrade subnet-evm on public deployment", func() {
+		commands.CreateSubnetEvmConfigWithVersion(subnetName, utils.SubnetEvmGenesisPath, subnetEVMVersion1)
+
+		// Simulate fuji deployment
+		s := commands.SimulateDeploySubnetPublicly(subnetName, keyName, controlKeys)
+		subnetID, _, err := utils.ParsePublicDeployOutput(s)
+		gomega.Expect(err).Should(gomega.BeNil())
+		// add validators to subnet
+		nodeInfos, err := utils.GetNodesInfo()
+		gomega.Expect(err).Should(gomega.BeNil())
+		for _, nodeInfo := range nodeInfos {
+			start := time.Now().Add(time.Second * 30).UTC().Format("2006-01-02 15:04:05")
+			_ = commands.SimulateAddValidatorPublicly(subnetName, keyName, nodeInfo.ID, start, "24h", "20")
+		}
+		// join to copy vm binary and update config file
+		for _, nodeInfo := range nodeInfos {
+			_ = commands.SimulateJoinPublicly(subnetName, nodeInfo.ConfigFile, nodeInfo.PluginDir)
+		}
+		// get and check whitelisted subnets from config file
+		var whitelistedSubnets string
+		for _, nodeInfo := range nodeInfos {
+			whitelistedSubnets, err = utils.GetWhilelistedSubnetsFromConfigFile(nodeInfo.ConfigFile)
+			gomega.Expect(err).Should(gomega.BeNil())
+			whitelistedSubnetsSlice := strings.Split(whitelistedSubnets, ",")
+			gomega.Expect(whitelistedSubnetsSlice).Should(gomega.ContainElement(subnetID))
+		}
+		// update nodes whitelisted subnets
+		err = utils.RestartNodesWithWhitelistedSubnets(whitelistedSubnets)
+		gomega.Expect(err).Should(gomega.BeNil())
+		// wait for subnet walidators to be up
+		err = utils.WaitSubnetValidators(subnetID, nodeInfos)
+		gomega.Expect(err).Should(gomega.BeNil())
+
+		// TODO Delete this after updating
+		var originalHash string
+
+		// upgrade the vm on each node
+		vmid, err := anr_utils.VMID(subnetName)
+		gomega.Expect(err).Should(gomega.BeNil())
+
+		for _, nodeInfo := range nodeInfos {
+			// check the current node version
+			vmVersion, err := utils.GetNodeVMVersion(nodeInfo.URI, vmid.String())
+			gomega.Expect(err).Should(gomega.BeNil())
+			gomega.Expect(vmVersion).Should(gomega.Equal(subnetEVMVersion1))
+
+			originalHash, err = utils.GetFileHash(filepath.Join(nodeInfo.PluginDir, vmid.String()))
+			gomega.Expect(err).Should(gomega.BeNil())
+		}
+
+		// stop network
+		commands.StopNetwork()
+
+		for _, nodeInfo := range nodeInfos {
+			output, err := commands.UpgradeVMPublic(subnetName, subnetEVMVersion2, nodeInfo.PluginDir)
+			gomega.Expect(err).Should(gomega.BeNil())
+			fmt.Println(output)
+			if err != nil {
+				fmt.Println(output)
+			}
+		}
+
+		// TODO: There is currently only one subnet-evm version compatible with avalanchego. These
```

Comment:
The test currently changes the binary to an old version and checks that the file changes. Your suggestion would break the test because the file would be the same. (The tool actually wouldn't perform the update because it would detect the version matches)